setup.py

```diff
@@ -261,6 +260,19 @@
 import setuptools.command.sdist
 from setuptools import Extension, find_packages, setup
 from setuptools.dist import Distribution
+
+
+cwd = os.path.dirname(os.path.abspath(__file__))
+
+# Add the current directory to the Python path so that we can import `tools`
+sys.path.insert(0, cwd)
+os.environ["PYTHONPATH"] = os.pathsep.join(
+    [
+        cwd,
+        os.getenv("PYTHONPATH", ""),
+    ]
+).rstrip(os.pathsep)
```

Comment:
Why this is relevant now, but was not before that? Also, why add it to PYTHON_PATH?

Reply:
I added a comment above the code to explain this.

From [PEP 517 – A build-system independent format for source trees](https://peps.python.org/pep-0517) documentation:

> When importing the module path, we do *not* look in the directory containing the source tree, unless that would be on `sys.path` anyway (e.g. because it is specified in `PYTHONPATH`).

------

> Why this is relevant now, but was not before that?

Previously, we were using the legacy build backend, which allowed importing local modules.

------

> Also, why add it to PYTHON_PATH?

I can build from source successfully without setting `PYTHONPATH`. But I think it would be better to set it so make the subprocess can look up the local build directory.